### PR TITLE
Tpetra: (local map) device default constructor fix for #11186

### PR DESCRIPTION
**Pull request by romintomasetti**
@trilinos/Tpetra

Suggested reviewer: @brian-kelley 

## Motivation

`Tpetra::Details::LocalMap` and `Tpetra::Details::FixedHashTable` had no device default constructor, leading to issues when using them in a `Kokkos::DualView` with `cuda uvm`. See related issue for more information.

## Related Issues

* Related to #11186

## Testing

We tested the patch for both `cuda` and `rocm`.

### CODE REVIEW

**Comment by brian-kelley** [comment]
@romintomasetti If you have code lying around that you used to test this, would you mind adding it to ``packages/tpetra/core/test/Map/Map_LocalMap.cpp`` as a unit test? Just something basic that exercises a View of LocalMaps in device code.

**Comment by romintomasetti** [comment]
@brian-kelley I've added a test, tell me if it's ok for you ;)

**Review by brian-kelley — APPROVED** [review]
Thanks for adding a test. The changes look good.

I'm also watching this discussion on slack since I want to understand this behavior better. Can you hold off on merging until that's resolved, just in case there's something that needs to change?

**Comment by romintomasetti** [comment]
Hi @brian-kelley !
The Slack discussion was summarized in #11247.

**Comment by brian-kelley** [comment]
@romintomasetti Thanks! I just marked this as inspected so that the autotester will run, and then you can merge this once it passes.

**Comment by brian-kelley** [comment]
@romintomasetti I looked at what was causing the failures and it turned out to be pretty straightforward (but likely painful to figure out if you haven't been a Trilinos developer for a long time :smile: ), so I just opened a PR against your branch here: https://github.com/uliegecsm/Trilinos/pull/4 . With it I can build Tpetra for Cuda or Serial and all the tests pass.

The main change was just that the ``FixedHashTable()`` default constructor definition needed to go into the decl header so that it could be compiled inline in device code.

**Review by brian-kelley — APPROVED** [review]

**Pull request merged**
